### PR TITLE
Core: Don't fail if catalog doesn't support views

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java [hunk 1]
```diff
@@ -717,8 +718,15 @@ public Transaction createTransaction() {
...
-        throw new AlreadyExistsException("View with same name already exists: %s", ident);
+      try {
+        if (viewExists(context, ident)) {
+          throw new AlreadyExistsException("View with same name already exists: %s", ident);
+        }
+      } catch (RESTException | UnsupportedOperationException e) {
+        // don't fail if the server doesn't support views, which could be due to:
+        // 1. server or backing catalog doesn't support views
+        // 2. newer client talks to an older server that doesn't support views
+        LOG.debug("Could not check whether view {} exists: {}", ident, e.getMessage(), e);
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java:729]: Style: typically we would use "Failed to check whether view {} exists" and leave the message to the exception trace.

Comment (nastra) [reply core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java]: updated

**Review by amogh-jahagirdar — APPROVED** [review]

File: core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java [hunk 2]
```diff
@@ -717,8 +718,15 @@ public Transaction createTransaction() {
...
-        throw new AlreadyExistsException("View with same name already exists: %s", ident);
+      try {
+        if (viewExists(context, ident)) {
+          throw new AlreadyExistsException("View with same name already exists: %s", ident);
+        }
+      } catch (RESTException | UnsupportedOperationException e) {
+        // don't fail if the server doesn't support views, which could be due to:
+        // 1. server or backing catalog doesn't support views
+        // 2. newer client talks to an older server that doesn't support views
+        LOG.debug("Could not check whether view exists: {}", ident, e);
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java]: I've updated this to `Failed to check whether view {} exists`

**Comment by rdblue** [comment]
Thanks, @nastra!

**Pull request merged**